### PR TITLE
New C# template for getting a Storage SAS token

### PR DESCRIPTION
**Pull request by lindydonna**

### CODE REVIEW

File: Templates/SasToken-CSharp/function.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "disabled": false,
+    "bindings": [
+        {
+            "authLevel": "function",
+            "name": "req",
+            "type": "httpTrigger",
+            "direction": "in"
```
Comment (mathewc) [inline Templates/SasToken-CSharp/function.json]: What verbs do you support? If it's just POST, recommend specifying `"methods": [ "post" ]' here

Comment (lindydonna) [inline Templates/SasToken-CSharp/function.json]: Sure. What does that affect?

Comment (mathewc) [inline Templates/SasToken-CSharp/function.json]: It just restricts the verbs you support. If requests come in for unsupported verbs, a 405 "Method Not Allowed" will be returned

**Comment by lindydonna** [comment]
@mathewc I've made the same updates to this sample. Could you please review?

File: Templates/SasToken-CSharp/run.csx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    if (data.container == null) {
+        return req.CreateResponse(HttpStatusCode.BadRequest, new {
+            error = "Specify value for 'container'"
+        });
+    }
+
+    var permissions = SharedAccessBlobPermissions.Read; // default to read permissions
+    Enum.TryParse(data.permissions.ToString(), out permissions);
+
+    var storageAccount = CloudStorageAccount.Parse(ConfigurationManager.AppSettings["AzureWebJobsStorage"]);
```
Comment (mathewc) [inline Templates/SasToken-CSharp/run.csx]: You could simplify this example by adding a `CloudStorageAccount account` parameter to your function signature. The WebJobs SDK will then pass it in for you automatically. Of course that only works if you're targeting the "AzureWebJobsStorage" default account.

File: Templates/SasToken-CSharp/run.csx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    string sasBlobToken;
+
+    // Get a reference to a blob within the container.
+    // Note that the blob may not exist yet, but a SAS can still be created for it.
+    CloudBlockBlob blob = container.GetBlockBlobReference(blobName);
+
+    if (policyName == null) {
+        var adHocSas = CreateAdHocSasPolicy(permissions);
+
+        //Generate the shared access signature on the blob, setting the constraints directly on the signature.
```
Comment (mathewc) [inline Templates/SasToken-CSharp/run.csx]: nit: add a space after //

File: Templates/SasToken-CSharp/run.csx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    dynamic data = await req.Content.ReadAsAsync<object>();
+
+    if (data.container == null) {
+        return req.CreateResponse(HttpStatusCode.BadRequest, new {
+            error = "Specify value for 'container'"
+        });
+    }
+
+    var permissions = SharedAccessBlobPermissions.Read; // default to read permissions
+    Enum.TryParse(data.permissions.ToString(), out permissions);
```
Comment (mathewc) [inline Templates/SasToken-CSharp/run.csx]: You should use the result of the TryParse - if it fails, return a 400?

**Comment by mathewc** [comment]
Just a few minor comments, otherwise looks good.

File: Templates/SasToken-CSharp/run.csx (hunk 2 above)
Comment (lindydonna) [inline Templates/SasToken-CSharp/run.csx]: Ah, good tip!  I think I'll leave it, so that customers can use a different app setting.

**Comment by lindydonna** [comment]
@mathewc I did a new commit so you could see the diff easily. If it looks ok, I'll squash them.

**Comment by mathewc** [comment]
:ship

**Pull request merged**